src/decode.rs

```diff
@@ -4526,15 +4545,12 @@ pub(crate) unsafe fn rav1d_decode_frame_init(
             return Err(ENOMEM);
         }
         if c.n_fc > 1 {
-            freep(&mut f.frame_thread.b as *mut *mut Av1Block as *mut c_void);
             freep(&mut f.frame_thread.cbi as *mut *mut CodedBlockInfo as *mut c_void);
-            f.frame_thread.b =
-                malloc(::core::mem::size_of::<Av1Block>() * num_sb128 as usize * 32 * 32)
-                    as *mut Av1Block;
+            f.frame_thread.b = vec![Default::default(); num_sb128 as usize * 32 * 32].into(); // TODO: fallible allocation
```

Comment:
```suggestion
            f.frame_thread.b.resize_with(num_sb128 as usize * 32 * 32, Default::default);
```

If `b` is a `Vec<Av1Block>` instead, this cuts down on some of the extra work we have to do here in default initializing all of the elements for safety, as we only need to default initialize new elements.  Also, this reuses the allocation, something the C didn't do, but is very reasonable to do.  Also, by doing it this way, we can remove the `#[derive(Clone, Copy)]` on `Av1Block` (32 bytes) and `Av1Block_intra_inter`.

Reply:
Hrm, that wouldn't reset the existing elements of the array though right? The original version was resetting all the elements, whereas using `resize_with` would only give you default values for new elements that were added. But we could do `clear` first and then `resize_with`, which should have the same effect. I'm also in favor of reusing the allocation, so seems like a good change to make 👍

Reply:
The C code didn't initialize the new elements when it allocates.  But in Rust, we have to initialize them for safety.  But it doesn't matter what value we initialize them to.  Your way default initializes them, but we could instead only default initialize new elements.  For existing elements, they can be initialized to whatever old values were there, no matter what they are.  So we save on the cost of having to default initialize everything, and only do so for new elements, which is more similar to how C doesn't initialize anything.

Reply:
I removed the `clear` call and left just `resize_with` 👍